### PR TITLE
Set serial console configuration for grub for ec2 firmware (bsc#1071135)

### PR DESCRIPTION
**Pull request by davidcassany**
This commit sets the serial console for grub2 in case an ec2 firmware
is used. This is a hack to enable serial console for Azure images.

### CODE REVIEW

File: modules/KIWIBoot.pm [hunk 1]
```diff
@@ -4608,6 +4608,12 @@ sub setupBootLoaderConfiguration {
         # console (console) or the serial terminal (serial)
         my $console_mode = 'graphics';
 
+        # Set serial console mode for ec2* firmwares. This is a hack for
+        # Azure images in order to avoid new attributes in XML schema.
+        if ($type->{firmware} =~ m/ec2/) {
```
Comment (rjschwei) [inline modules/KIWIBoot.pm]: Also needs the condition for $type->{format} eq 'vhdfixed'

Then we catch both Azure and EC2

File: modules/KIWILinuxRC.sh [hunk 2]
```diff
@@ -3162,6 +3162,14 @@ EOF
     else
         echo "GRUB_TERMINAL=console"  >> $inst_default_grub
     fi
+
+    # Set serial console mode for ec2* firmwares. This is a hack for
+    # Azure images in order to avoid new attributes in XML schema.
+    if [[ "$kiwi_firmware" =~ "ec2" ]]; then
```
Comment (rjschwei) [inline modules/KIWILinuxRC.sh]: Also needs the format check for vhdfixed.

Comment (davidcassany) [reply modules/KIWILinuxRC.sh]: Here we do not have any $kiwi_format variable available, checking how it can be included here is needed for this change.

**Comment by davidcassany** [comment]
@rjschwei would it be possible for you test it? I am having issues to boot the image with qemu and so far I could just check the disk contains the appropriate grub2.cfg and that the .profile file includes the format for the boot code.

I have no experience with ec2 and I have no idea how should I test it in my local env.

**Comment by davidcassany** [comment]
I have added block label to make sure it is not merged. The patch is not sufficient, grub is not configured properly after the first boot. The boot code sets the /etc/default/grub file correclty so I can see the GRUB_SERIAL_COMMAND variable with the correct values; however after running the grub2-mkconfig (it is executed by the boot code on a later stage) the resulting file does not contain this information.

**Comment by davidcassany** [comment]
Right, finally found the that GRUB_SERIAL_COMMAND is only applied when GRUB_TERMINAL is set to serial. Now I have successfully tested in ec2 firmware and with vhd-fixed format.

**Comment by davidcassany** [comment]
@rjschwei, @schaefi when you approve the PR I will proceed with the maintenance update. 

Thanks

**Comment by schaefi** [comment]
Looks great 👍

**Pull request merged**